src/parser/peg.rs

```diff
@@ -36,9 +45,17 @@ impl Pipeline {
             job.expand(&expanders);
         }
 
-        if let Some(stdin) = self.stdin.iter_mut().next() {
-            stdin.file = expand_string(stdin.file.as_str(), &expanders, false).join(" ");
-        }
+        let stdin = match self.stdin {
+            Some(Input::File(ref s)) => {
+                Some(Input::File(expand_string(s, &expanders, false).join(" ")))
+            },
+            Some(Input::HereString(ref s)) => {
+                Some(Input::HereString(expand_string(s, &expanders, false).join(" ")))
```

Comment:
If you change the `false` to `true` here, the newlines won't be replaced with spaces.